internal/utils.go

```diff
@@ -82,3 +85,54 @@ func CleanupTasks(tasks []Task) {
 		}
 	}
 }
+
+func CreateTarWithFiles(tarPath string, filePaths []string) error {
```

Comment:
This function is being used exclusively for tests - I don't think it belongs in this file since it is not a general utility function